### ISSUE TITLE
Nan in training data

### ISSUE THREAD

**Issue by aragakiyui611** [EricGuo5513/momask-codes#25]
when I run `python train_vq.py --name rvq_name --gpu_id 1 --dataset_name t2m --batch_size 256 --num_quantizers 6  --max_epoch 50`, it prints this bug, is there are NaNs in the datasets?
<img width="543" alt="无标题" src="https://github.com/EricGuo5513/momask-codes/assets/74336517/a7f9afc1-ac13-4818-a06e-002ada38f3df">

**Comment by Murrol** [comment]
Hi,

There should not be any NaN in HumanML3D. Please refer to the [tutorial](https://github.com/EricGuo5513/HumanML3D/blob/main/motion_representation.ipynb) to check if your data is correct. Specifically, the error message should refer to the validation set. I suggest you check if you processed the dataset correctly. If it's overall correct, then just filter out the NaN sequence for a quick fix.

**Comment by EricGuo5513** [comment]
You may also check if you have matched numpy and scipy library version.

**Comment by aragakiyui611** [comment]
I regenerate the dataset and it still has nan. I tried to filter them out.
![image](https://github.com/EricGuo5513/momask-codes/assets/74336517/fc40ad42-f3b0-4d10-8289-1b1d1cc6d98a)

However there still some data problem.
<img width="545" alt="无标题" src="https://github.com/EricGuo5513/momask-codes/assets/74336517/222a585c-e2ea-4094-83c3-2b519d5dca7b">

**Comment by Murrol** [comment]
There’s a problem in your current filter solution that you ignored the constraints of the covariance matrix. E.g. the diagonal elements must be larger than 0.

What I suggest for a quick fix is to check the motion sequences in the dataset. Hopefully, there would be only few illegal sequences. Then, filter out all the illegal sequences before any operations. I also suggest you check which operation involves NaN data but not mess them up.

**Comment by aragakiyui611** [comment]
Thank you！ I found the data have nan and deleted them. There were only *007975.npy has nan.
